src/java/org/apache/cassandra/config/Config.java

```diff
@@ -56,17 +55,24 @@
     public String authorizer;
     public String role_manager;
     public String network_authorizer;
-    public volatile int permissions_validity_in_ms = 2000;
+
+    @Replaces(oldName = "permissions_validity_in_ms", converter = Converter.MillisDurationConverter.class, deprecated = true)
+    public volatile CassandraDuration permissions_validity = new CassandraDuration("2s");
     public volatile int permissions_cache_max_entries = 1000;
-    public volatile int permissions_update_interval_in_ms = -1;
+    @Replaces(oldName = "permissions_update_interval_in_ms", converter = Converter.MillisDurationConverterCustom.class, deprecated = true)
+    public volatile CassandraDuration permissions_update_interval = new CassandraDuration("0ms");
     public volatile boolean permissions_cache_active_update = false;
-    public volatile int roles_validity_in_ms = 2000;
+    @Replaces(oldName = "roles_validity_in_ms", converter = Converter.MillisDurationConverter.class, deprecated = true)
+    public volatile CassandraDuration roles_validity = new CassandraDuration("2s");
     public volatile int roles_cache_max_entries = 1000;
-    public volatile int roles_update_interval_in_ms = -1;
+    @Replaces(oldName = "roles_update_interval_in_ms", converter = Converter.MillisDurationConverterCustom.class, deprecated = true)
+    public volatile CassandraDuration roles_update_interval= new CassandraDuration("0ms");
     public volatile boolean roles_cache_active_update = false;
-    public volatile int credentials_validity_in_ms = 2000;
+    @Replaces(oldName = "credentials_validity_in_ms", converter = Converter.MillisDurationConverter.class, deprecated = true)
+    public volatile CassandraDuration credentials_validity = new CassandraDuration("2s");
     public volatile int credentials_cache_max_entries = 1000;
-    public volatile int credentials_update_interval_in_ms = -1;
+    @Replaces(oldName = "credentials_update_interval_in_ms", converter = Converter.MillisDurationConverterCustom.class, deprecated = true)
+    public volatile CassandraDuration credentials_update_interval= new CassandraDuration("0ms");
```

Comment:
The default literally changes here from `-1` to `0ms` does that have functional consequences? (ex. We check for `null` in `AuthConfig` where we used to check for `-1`, but the default isn't `null`...)

Reply:
Bug in AuthConfig fixed